File: app/src/main/java/com/nbennettsoftware/android/npad/dialog/WarnUnsavedChangesDialog.java
package com.nbennettsoftware.android.npad.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatDialogFragment;

import com.nbennettsoftware.android.npad.R;

public class WarnUnsavedChangesDialog extends AppCompatDialogFragment {

    private OnWarningFinished onWarningFinished;

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage("Any unsaved changes to the document will be lost.");
        builder.setPositiveButton("Ok", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                if(onWarningFinished !=null) {
                    onWarningFinished._continue();
                }
            }
        });
        builder.setNeutralButton("Cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.cancel();
            }
        });
        setCancelable(false);
        return builder.create();
    }

    public interface OnWarningFinished {
        void _continue();
    }

    public void setOnWarningFinished(OnWarningFinished onWarningFinished){
        this.onWarningFinished = onWarningFinished;
    }


}

File: app/src/main/java/com/nbennettsoftware/android/npad/dialog/WarnUnsavedChangesDialog.kt
package com.nbennettsoftware.android.npad.dialog

import android.app.Dialog
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatDialogFragment

class WarnUnsavedChangesDialog : AppCompatDialogFragment() {

    var onWarningFinished: (()->Unit)? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!)
        builder.setMessage("Any unsaved changes to the document will be lost.")
        builder.setPositiveButton("Ok") { _, _ ->
            onWarningFinished?.invoke()
        }
        builder.setNeutralButton("Cancel") { dialog, _ -> dialog.cancel() }
        isCancelable = false
        return builder.create()
    }

}
